plugins/ufm_log_analyzer_plugin/src/loganalyze/log_analyzers/link_flapping_analyzer.py

```diff
@@ -0,0 +1,125 @@
+        self._mapped_samples = {"5m":{}, "1h":{}, "1d":{}, "1w":{}}
+        self._map_files_by_sampling_rate_and_date()
+        self._funcs_for_analysis={self.plot_link_flapping_last_week}
+
+    def _map_files_by_sampling_rate_and_date(self):
+        for sample_file in self._telemetry_samples_csv:
+            try:
+                file_name = os.path.basename(sample_file)
+                match = re.match(FILE_NAME_PATTERN, file_name)
+                if match:
+                    interval_time = match.group(1)
+                    date_time_str = match.group(2)
+
+                    # Convert the date_time string to a datetime object
+                    date_time = datetime.strptime(date_time_str, TIME_PATTERN)
+                    self._mapped_samples[interval_time][date_time] = sample_file
+                else:
+                    log.LOGGER.debug(f"File {file_name} cannot be used for links flapping")
+            except FileNotFoundError:
+                log.LOGGER.debug(f"File {sample_file} does not exist")
+            except re.error:
+                log.LOGGER.debug(f"Invalid regular expression pattern {FILE_NAME_PATTERN}")
+            except ValueError:
+                log.LOGGER.debug(f"Invalid date/time format {date_time_str}")
+            except KeyError:
+                log.LOGGER.debug(f"Invalid interval time {interval_time}")
+        #Sorting the inner dict so the first item would be the last timestamp
+        sorted_data = {key: sorted(value.items(),
+                                   key=lambda x: x[0],
+                                   reverse=True) for key, value in self._mapped_samples.items()}
+        self._mapped_samples = {key: dict(value) for key, value in sorted_data.items()}
+
+    def _ungz_to_csv(self, file_path):
+        # Ensure the input file has a .gz extension
+        if not file_path.endswith('.gz'):
+            return ""
+
+        # Create the output file path by replacing .gz with .csv
+        output_file_path = f"{Path(file_path).stem}.csv"
+
+        # Open the .gz file and write its content to a new .csv file
+        with gzip.open(file_path, 'rb') as f_in:
+            with open(output_file_path, 'wb') as f_out:
+                shutil.copyfileobj(f_in, f_out)
+
+        return output_file_path
+
+    def _get_link_flapping_by_gz_files(self, file1_gz, file2_gz):
+        link_flapping = get_link_flapping(self._ungz_to_csv(file1_gz),
+                                          self._ungz_to_csv(file2_gz))
+        columns_to_keep =['link_hash_id', 'estimated_time']
+        link_flapping = link_flapping.loc[:, columns_to_keep]
+        # Drop duplicate columns
+        link_flapping = link_flapping.loc[:, ~link_flapping.columns.duplicated()]
+        # Reset the index and drop the old index column
+        link_flapping = link_flapping.reset_index(drop=True)
+        return link_flapping
+
+    def get_link_flapping_last_week(self):
```

Comment:
can we maybe rename it to just "get_link_flapping" and pass an interval as a parameter? just thinking that one day we could change it from 1w to 2w, and in this case we will have to rename all of it.

Reply:
The reason I gave it this name is that I did not make it configurable.
This step is to see if it has any value.

At first, I added it as a generic with default.
But there are cases where you might not get the 2 weeks old samples or other things.

I prefer to look at it as a code smell.
This is the first time, so let's see where it goes.we might get to the conclusion this should be deleted :( 


If after my explanation you still think the name should be changed and we need to make it configurable, I'll do that (with now supporting only 1 week)

Reply:
ok, I trust you on that one, let's leave it like this.